assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -292,6 +356,47 @@ const baseReducer = ( state, { type, payload } ) => {
 				},
 			};
 		}
+		case SET_SETTINGS_COMPONENT: {
+			return {
+				...state,
+			};
+		}
+		case SET_SETTINGS_DISPLAY_MODE: {
+			const { slug, status } = payload;
+			const { modules: existingModules } = state;
+
+			// Clone object so as not to override previous object's values below.
+			const updatedModules = JSON.parse( JSON.stringify( existingModules ) );
```

Comment:
I think we can avoid using `JSON` functions here. How about we create the object as `updatedModules = { ...existingModules }` and then below, wherever we modify a module we use `updatedModules[ slug ] = { ...updatedModules[ slug ], displayMode: '...' }`.